opensearch-project/OpenSearch-Dashboards
# [MD] Concatenate data source name with index pattern name and change delimiter to double colon

### Description
This change adds data source name as a prefix to the index pattern name for the index pattern selector. This change also changed the delimiter for the datasource name and index pattern name to be double colons by changing getIndexPatternTitle function which is used by index pattern service(https://github.com/opensearch-project/OpenSearch-Dashboards/blob/eff7cb5536a552692c43fedd60fb8a9f87f1e801/src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts#L127) and saved object management page(https://github.com/opensearch-project/OpenSearch-Dashboards/blob/eff7cb5536a552692c43fedd60fb8a9f87f1e801/src/plugins/saved_objects_management/server/routes/find.ts#L99). 

### Issues Resolved

Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5900

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/fc52f753-dbde-4a5f-974d-b2e5818f6779



## Testing the changes
The following steps were performed to test the change in the recording:
1. disable data source, goes to maps plugin page
2. click `add layer`, then see the index pattern selector with options populated
3. go to visualization and select `Controls` , then click `Add`, will see index pattern selector with options populated correctly
4. enable data source, goes to the same visualization page, and will see the options that have data source name prefixed before index pattern name, the delimiter is double colon
5. go to maps plugin, click `add layer`, then see the index pattern selector with options populated, data source name is added as prefix before the index pattern name and the delimiter is double colon

### Check List
